src/Config/GlobalQuery.ts

```diff
@@ -0,0 +1,32 @@
+import { Query } from '../Query/Query';
+
+export class GlobalQuery {
+    private static instance: GlobalQuery;
+
+    static empty = '';
+    private _source;
+
+    public static getInstance(): GlobalQuery {
+        if (!GlobalQuery.instance) {
+            GlobalQuery.instance = new GlobalQuery();
+        }
+
+        return GlobalQuery.instance;
+    }
+
+    constructor(source: string = GlobalQuery.empty) {
+        this._source = source;
+    }
+
+    public set(source: string) {
+        this._source = source;
+    }
+
+    public query(path: string | undefined = undefined): Query {
+        return new Query({ source: this._source }, path);
+    }
+
+    public isEmpty() {
```

Comment:
It's highly non-obvious that whitespace is ignored in this calculation - it would be more accurately named as `isEmptyOrPurelyWhitespace()`.

Failing that, please add JSDoc cautioning about this behaviour.

Reply:
I do appreciate that the trim arose from a previous code review. So just documenting the behaviour is sufficient for now.

Reply:
Indeed, this should be renamed and the method ignores not just whitespaces but also line breaks. So I gave it some thought, added jsdoc and renamed for it's intent - at the end of the day what this method says is whether the Global Query has any instructions so...- `hasInstructions()`.

This way the tests became much more descriptive:

```js
it.each(['', ' ', '\n', '\n     \n    ', '  \n    \n'])(
        'should have no instructions if line breaks and spaces were set in the query',
        (globalQuerySource) => {
            const globalQuery = new GlobalQuery(globalQuerySource);

            expect(globalQuery.hasInstructions()).toEqual(false);
        },
    );
```

Hope you share this point of view.

Reply:
> at the end of the day what this method says is whether the Global Query has any instructions so...- hasInstructions()

I like the new name. Thanks.